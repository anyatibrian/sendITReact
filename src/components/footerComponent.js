import React from "react";
const FooterComponent=(props)=>{
  return(
      <div className="footer-element">
        <div className="row">
          <div className="col-lg-3">
          </div>
          <div className="col-lg-6">
            <h6>
              @copy all rights reserved sendIT 2019
            </h6>
          </div>
          <div className="col-lg-3">
          </div>
        </div>
      </div>
  );
};
export default FooterComponent;