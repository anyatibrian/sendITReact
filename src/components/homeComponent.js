import React from "react";
export const  HomeComponent=(props)=>{
  const {
    toggleButton,
    display,
    receiversName,
    description,
    pickup,
    weight,
    destination,
    onChange,
    onSubmit
  }= props;
    return (
        <div className="home-container">
          <div className="row">
            <div className="col col-lg-1">
            </div>
            <div className="col col-lg-10">
              <button className="btn btn-success" onClick={toggleButton}>new order</button>
            </div>
            <div className="col col-lg-1">
          </div>
          </div>
          <div className="row">
            <div className="col col-lg-4">
            </div>
            <div className="col col-lg-4">
              {display &&
              <div className="modal" style={{display:display}}>
                <div className="">
                  <div className="row">
                    <div className="col col-lg-3">
                    </div>
                    <div className="col col-lg-6">
                      <div className="modal-content">
                        <span className="close" id="update_add_modal" onClick={toggleButton} >&times;</span>
                        <h2>New Parcel Order</h2>
                        <form method="POST" onSubmit={onSubmit}>
                          <div className="form-group">
                            <label>Receivers Name</label>
                            <input
                                type="text"
                                className="form-control"
                                name="receiversName"
                                required
                                value={receiversName}
                                onChange={onChange}
                            />
                          </div>
                          <div className="form-group">
                            <label>Description</label>
                            <textarea
                                className="form-control"
                                name="description"
                                required
                                value={description}
                                onChange={onChange}
                            >
                            </textarea>
                          </div>
                          <div className="form-group">
                            <label>Pickup</label>
                            <input
                                type="text"
                                className="form-control"
                                name="pickup"
                                required
                                value={pickup}
                                onChange={onChange}
                            />
                          </div>
                          <div className="form-group">
                            <label>Destination</label>
                            <input
                                type="text"
                                className="form-control"
                                name="destination"
                                required
                                value={destination}
                                onChange={onChange}
                            />
                          </div>
                          <div className="form-group">
                            <label>weight</label>
                            <input
                                type="number"
                                className="form-control"
                                name="weight"
                                required
                                value={weight}
                                onChange={onChange}
                            />
                          </div>
                          <div className="form-group">
                            <button className="btn btn-order">
                              Place Order
                            </button>
                          </div>
                        </form>
                      </div>
                    </div>
                    <div className="col col-lg-3">
                    </div>
                  </div>
                </div>
              </div>
              }
            </div>
            <div className="col col-lg-4">
            </div>
          </div>
        </div>
    );
};
export default HomeComponent