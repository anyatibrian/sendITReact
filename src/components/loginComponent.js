import React from "react";
import { Link } from "react-router-dom";
export const LoginComponent = (props) => {
  const {
    username,
    password,
    handleChange,
    handleSubmit,
    isLoading
  }= props;
  return (
    <div className="login-container">
      <div className="row mt-5">
        <div className="col col-lg-6 col-md-6" />
        <div className="col col-lg-5 col-md-3">
          <div className="">
            <form method="POST" onSubmit={handleSubmit} className="form">
              <h2 className="header-login"> Login </h2>
              <div className="form-group">
                <input
                  type="text"
                  className="form-control"
                  id="username"
                  name="username"
                  placeholder="username"
                  value={username}
                  onChange={handleChange}
                />
              </div>
              <div className="form-group">
                <input
                  type="password"
                  className="form-control"
                  id="password"
                  name="password"
                  placeholder="password"
                  value={password}
                  onChange={handleChange}
                />
              </div>
              <input
                type="submit"
                className="btn btn-success login-button"
                value={isLoading?"loading":"login"}
              />
              <p className="">
                <Link to="/register" id="Link">
                  Don't have an account yet, create one
                </Link>
              </p>
            </form>
          </div>
        </div>
        <div className="col col-lg-1 col-md-1" />
      </div>
    </div>
  );
};
