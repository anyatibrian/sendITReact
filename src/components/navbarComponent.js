import React, { Component } from "react";
import {connect} from "react-redux";
import "../sass/home.scss";
import LOGO from  "../sass/images/default.jpg";
class NavbarComponent extends Component {
  constructor(props) {
    super(props);
  }
  logout = () => {
    let {history} = this.props;
    sessionStorage.clear();
    history.push("/login");

  };
  render() {
    const {isLoggedIn} = this.props;
    return (
        <div>
           <div className="home-container">
            <nav className="navbar navbar-expand-lg navbar-light bg-light">
              <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="nav navbar-nav ml-auto">
                  <li className="nav-item">
                    <img src={LOGO} className="profile-picture"/>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="#" onClick={this.logout}>logout</a>
                  </li>
                </ul>
              </div>
            </nav>
          </div>
        </div>

    );
  }
}
const mapStateToProps=state=>({
  isLoggedIn:state.login.isLoggedIn
});
export default connect(mapStateToProps, null) (NavbarComponent);
