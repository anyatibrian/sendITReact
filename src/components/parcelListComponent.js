import React from "react";
export const  ParcelListComponent =(props)=>{
  return (
      <div className="home-container">
        <div className="main-content">
          <div className="row userAdmin-dashboard-content">
            <div className="col col-lg-1">
            </div>
            <div className="col col-lg-10">
              <table id="myTable">
                <tr className="header">
                  <th style={{width:"10%"}}>Serial No.</th>
                  <th style={{width:"10%"}}>Order By</th>
                  <th style={{width:"10%"}}>Parcel</th>
                  <th style={{width:"30%"}}>Description</th>
                  <th style={{width:"10%"}}>Pick up</th>
                  <th style={{width:"10%"}}>Destination</th>
                  <th style={{width:"10%"}}>Location</th>
                  <th style={{width:"15%"}}>Price</th>
                  <th style={{width:"10%"}}>Status</th>
                  <th style={{width:"15%"}}>Delivery</th>
                </tr>
                <tr>
                  <td>12856</td>
                  <td>Serwage</td>
                  <td>cooker</td>
                  <td>
                    Lorem ipsum, or lipsum as it is sometimes known, is dummy text used in laying out print, graphic or
                    web designs.
                  </td>
                  <td>Tororo</td>
                  <td>Kampala</td>
                  <td>Jinja</td>
                  <td>4600 Ugx</td>
                  <td>Transit</td>
                  <td>
                    <img src="https://img.icons8.com/color/48/000000/approve-and-update.png"
                         style={{width:"20px", height:"20px"}}/>
                  </td>
                </tr>
                <tr>
                  <td>12856</td>
                  <td>Akim</td>
                  <td>cooker</td>
                  <td>
                    Lorem ipsum, or lipsum as it is sometimes known, is dummy text used in laying out print, graphic or
                    web designs.
                  </td>
                  <td>Tororo</td>
                  <td>Kampala</td>
                  <td>Jinja</td>
                  <td>4600 Ugx</td>
                  <td>Delivered</td>
                  <td>
                    <img src="https://img.icons8.com/color/48/000000/approve-and-update.png"
                         style={{width:"20px", height:"20px"}}/>
                  </td>
                </tr>
                <tr>
                  <td>12856</td>
                  <td>Brayan</td>
                  <td>laptops</td>
                  <td>
                    Lorem ipsum, or lipsum as it is sometimes known, is dummy text used in laying out print, graphic or
                    web designs.
                  </td>
                  <td>Tororo</td>
                  <td>Kampala</td>
                  <td>Jinja</td>
                  <td>4600 Ugx</td>
                  <td>Transit</td>
                  <td>
                    <img src="https://img.icons8.com/color/48/000000/approve-and-update.png"
                         style={{width:"20px", height:"20px"}}/>
                  </td>
                </tr>
                <tr>
                  <td>12856</td>
                  <td>Rolyce</td>
                  <td>cooker</td>
                  <td>
                    Lorem ipsum, or lipsum as it is sometimes known, is dummy text used in laying out print, graphic or
                    web designs.
                  </td>
                  <td>Tororo</td>
                  <td>Kampala</td>
                  <td>Jinja</td>
                  <td>4600 Ugx</td>
                  <td>Transit</td>
                  <td>
                    <img src="https://img.icons8.com/color/48/000000/approve-and-update.png"
                         style={{width:"20px", height:"20px"}}/>
                  </td>

                </tr>
                <tr>
                  <td>12856</td>
                  <td>popo</td>
                  <td>cooker</td>
                  <td>
                    Lorem ipsum, or lipsum as it is sometimes known, is dummy text used in laying out print, graphic or
                    web designs.
                  </td>
                  <td>Tororo</td>
                  <td>Kampala</td>
                  <td>Jinja</td>
                  <td>4600 Ugx</td>
                  <td>Transit</td>
                  <td>
                    <img src="https://img.icons8.com/color/48/000000/approve-and-update.png"
                         style={{width:"20px", height:"20px"}}/>
                  </td>
                </tr>
              </table>
            </div>
            <div className="col col-lg-1">
            </div>
          </div>
      </div>
      </div>
  );
};
export default ParcelListComponent;