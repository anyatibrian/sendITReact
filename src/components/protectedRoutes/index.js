import React from "react";
import { Redirect, Route } from "react-router-dom";
import decode from "jwt-decode";

export const Authenticate = (token) => {
  try {
    const res = decode(token);
    if (res.exp > Date.now() / 1000) {
      return { res };
    }
    return false;
  } catch (error) {
    return false;
  }
};

const ProtectedRoute = ({ component: Component, ...props }) => {
  const isLoggedIn = Authenticate(sessionStorage.getItem("token"));
  return (
      <Route
          {...props}
          render={props => (isLoggedIn ? <Component {...props} /> : <Redirect to="/login" />)
          }
      />
  );
};

export default ProtectedRoute;