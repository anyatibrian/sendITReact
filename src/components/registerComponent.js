import React from "react";
import { Link } from "react-router-dom";
const RegisterComponent = props => {
  const {
    errors,
    username,
    email,
    password,
    confirmPassword,
    onChange,
    onSubmit,
    hidden,
    onToggle,
    isLoading
  } = props;
  return (
    <div className="row mt-5">
      <div className="col col-lg-6 col-md-5" />
      <div className="col col-lg-5 col-md-6">
        <div className="">
          <form onSubmit={onSubmit} className="form">
            <h2 className="header-login"> Register </h2>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                name="username"
                id="username"
                value={username}
                onChange={onChange}
                placeholder="username"
              />
              <p>
                {errors.username.length > 0 ? (
                  <span> {errors.username}</span>
                ) : (
                  ""
                )}
              </p>
            </div>
            <div className="form-group">
              <input
                type="text"
                name="email"
                className="form-control"
                value={email}
                onChange={onChange}
                id="email"
                placeholder="johdoe@gmail.com"
              />
              <p>
                {errors.email.length > 0 ? <span>{errors.email}</span> : ""}
              </p>
            </div>
            <div className="form-group">
              <input
                type={hidden ? "text" : "password"}
                name="password"
                className="form-control"
                value={password}
                id="password"
                onChange={onChange}
                placeholder="password"
                onClick={onToggle}
              />
              {errors.password.length > 0 ? (
                <span style={{ color: "white" }}>{errors.password}</span>
              ) : (
                ""
              )}
            </div>
            <div className="form-group">
              <input
                type="password"
                name="confirmPassword"
                className="form-control"
                value={confirmPassword}
                id="passwordConfirm"
                onChange={onChange}
                placeholder="passwordConfirm"
              />
              {errors.confirmPassword.length > 0 ? (
                <span style={{ color: "white" }}>{errors.confirmPassword}</span>
              ) : (
                ""
              )}
            </div>

            <input
              type="submit"
              className="btn btn-success login-button"
              value={isLoading?"loading....":"register"}
            />
            <p className="">
              <Link to="/" id="Link">
                Already have an account ,login
              </Link>
            </p>
          </form>
        </div>
      </div>
      <div className="col col-lg-1" />
    </div>
  );
};
export default RegisterComponent;
