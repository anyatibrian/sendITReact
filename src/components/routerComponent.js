import React, { Component } from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import LoginContainer from "../containers/loginContainer";
import NavbarComponent from "../components/navbarComponent";
class RouterComponent extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <BrowserRouter>
        <NavbarComponent/>
        <Switch>
          <Route path="/login" component={LoginContainer}/>
        </Switch>
      </BrowserRouter>
    );
  }
}
export default RouterComponent;