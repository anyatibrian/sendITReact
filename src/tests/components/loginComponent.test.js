import React from "react";
import {shallow} from "enzyme";
import {LoginComponent} from "../../components/loginComponent";

let wrapper;
let props = {
  username:"",
  password:"",
  handleChange:jest.fn(),
  handleSubmit:jest.fn(),
  isLoading:""
};
describe('LoginComponent', ()=>{
  beforeEach(()=>{
    wrapper = shallow(<LoginComponent {...props}/>);
  });
  it('should render without crashing', ()=>{
    wrapper.setProps({isLoading:"loading"});
    expect(wrapper).toMatchSnapshot();
  });
});
