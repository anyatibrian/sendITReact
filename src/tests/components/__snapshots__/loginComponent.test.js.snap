// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should render without crashing 1`] = `ShallowWrapper {}`;
